### ISSUE TITLE
Dataset Clustering and Re-split

### ISSUE THREAD

**Issue by fire717** [HighwayWu/DeepFakeDefenders#11]
作者您好，感谢开源！
我想问下，在你们的方案中，据描述是先对图片先提取特征然后无监督聚类，然后在每一个类别中再重新划分train、val是吧。那么有两个问题请教：

1. 提取特征是直接用冻结的CLIP来提取吗，然后用kmens直接聚类？
2. 比赛原始数据有train、val，那么你们是把这两个合并一起聚类了，再重新划分，还是说分别在train和val里单独聚类呢？如果是前者，会不会存在val的数据泄露到train里的情况？如果是后者，如果把train聚的类别和val聚的类别关联上呢？

谢谢！

**Comment by HighwayWu** [comment]
Thanks for your asking.
- We first train a small network (such as Eff-B0) based on binary hard labels, and then use it to extract latent features for all training data, resulting in an N*1024 matrix. Then, we perform unsupervised clustering (such as K-means) and subsequent re-split based on the clustering results.
- The CLIP model was not used because the competition restricted the use of ImageNet pre-trained models. However, we believe that CLIP (especially after fine-tuning) can achieve better data clustering.
- We cluster the training data directly and re-split it into new training and validation sets. This process does not include the official (original) validation set.

**Comment by fire717** [comment]
感谢分享，非常棒的工作！
